subsys/net/l2/ppp/lcp.c

```diff
@@ -203,6 +203,100 @@ static void lcp_finished(struct ppp_fsm *fsm)
 	ppp_link_terminated(ctx);
 }
 
+#if defined(CONFIG_PPP_DIALUP)
+
+#define MRU_OPTION_LEN 4
+#define MRU_DATA_LEN 2
+
+static int lcp_add_mru(struct ppp_context *ctx, struct net_pkt *pkt)
+{
+	uint8_t data[MRU_DATA_LEN];
+	
+	sys_put_be16(ctx->lcp.my_options.mru, data);
+
+	net_pkt_write_u8(pkt, MRU_OPTION_LEN);
+	return net_pkt_write(pkt, data, sizeof(data));
+}
+
+static int lcp_ack_mru(struct ppp_context *ctx, struct net_pkt *pkt,
+			       uint8_t oplen)
+{
+	int ret = 0;
+	uint16_t mru = 0;
+
+	/* Handle ACK : */
+	if (oplen != sizeof(mru)) {
+		return -EINVAL;
+	}
+
+	ret = net_pkt_read(pkt, &mru, sizeof(mru));
+	if (ret) {
+		return ret;
+	}
+	if (mru != ctx->lcp.my_options.mru) {
+		/* Didn't acked our MRU: */
+		return -EINVAL;
+	}
+
+	return 0;
+}
+
+static int lcp_nak_mru(struct ppp_context *ctx, struct net_pkt *pkt,
+			       uint8_t oplen)
+{
+	int ret = 0;
+	uint16_t mru = 0;
+
+	/* Handle NAK: accept only smaller/equal than ours */
+	if (oplen != sizeof(mru)) {
+		return -EINVAL;
+	}
+
+	ret = net_pkt_read(pkt, &mru, sizeof(mru));
+	if (ret) {
+		return ret;
+	}
+
+	if (mru <= ctx->lcp.my_options.mru) {
+		/* OK, reset the MRU also in our side: */
+		ctx->lcp.my_options.mru = mru;
+	} else {
+		return -EINVAL;
+	}
+	
+	return 0;
+}
+
+static const struct ppp_my_option_info lcp_my_options[] = {
+	PPP_MY_OPTION(LCP_OPTION_MRU, lcp_add_mru, lcp_ack_mru, lcp_nak_mru),
+};
+BUILD_ASSERT(ARRAY_SIZE(lcp_my_options) == LCP_NUM_MY_OPTIONS);
```

Comment:
Why do we need this build assert check?

Reply:
This is similarly than in IPCP options side for checking that those are in synch and noticing that editing of either one impacts on others.